File: crew/crew.go
package crew

import (
	"fmt"
	"time"
)

// CrewMember represents a member of the BMAD crew
type CrewMember struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Role      string    `json:"role"`
	Active    bool      `json:"active"`
	JoinedAt  time.Time `json:"joined_at"`
}

// Crew manages a collection of crew members
type Crew struct {
	Members []CrewMember `json:"members"`
}

// NewCrew creates a new empty crew
func NewCrew() *Crew {
	return &Crew{
		Members: make([]CrewMember, 0),
	}
}

// AddMember adds a crew member to the crew
func (c *Crew) AddMember(member CrewMember) {
	if member.ID == "" {
		member.ID = fmt.Sprintf("crew-%d", time.Now().UnixNano())
	}
	if member.JoinedAt.IsZero() {
		member.JoinedAt = time.Now()
	}
	c.Members = append(c.Members, member)
}

// RemoveMember removes a crew member by ID
func (c *Crew) RemoveMember(id string) bool {
	for i, m := range c.Members {
		if m.ID == id {
			c.Members = append(c.Members[:i], c.Members[i+1:]...)
			return true
		}
	}
	return false
}

// GetMember returns a crew member by ID
func (c *Crew) GetMember(id string) (*CrewMember, bool) {
	for _, m := range c.Members {
		if m.ID == id {
			return &m, true
		}
	}
	return nil, false
}

// ActiveMembers returns only active crew members
func (c *Crew) ActiveMembers() []CrewMember {
	active := make([]CrewMember, 0)
	for _, m := range c.Members {
		if m.Active {
			active = append(active, m)
		}
	}
	return active
}

File: src/bmad_crew/crew.py
"""BMAD Crew management module."""

from dataclasses import dataclass, field
from typing import List, Optional
from enum import Enum


class CrewStatus(Enum):
    """Possible states for a crew."""
    INACTIVE = "inactive"
    ACTIVE = "active"
    ON_HOLIDAY = "on_holiday"
    DISBANDED = "disbanded"


@dataclass
class CrewMember:
    """Represents a single member of a BMAD crew."""
    name: str
    role: str

    def __str__(self) -> str:
        return f"{self.name} ({self.role})"


@dataclass
class Crew:
    """Manages a BMAD crew with members and status."""
    name: str
    status: CrewStatus = CrewStatus.INACTIVE
    _members: List[CrewMember] = field(default_factory=list, repr=False)

    def add_member(self, member: CrewMember) -> None:
        """Add a member to the crew."""
        if member not in self._members:
            self._members.append(member)

    def remove_member(self, name: str) -> bool:
        """Remove a member by name. Returns True if found and removed."""
        for i, m in enumerate(self._members):
            if m.name == name:
                self._members.pop(i)
                return True
        return False

    def update_status(self, status: CrewStatus) -> None:
        """Update the crew's status."""
        self.status = status

    def roster(self) -> List[str]:
        """Return a list of member names in the crew."""
        return [m.name for m in self._members]

    def __str__(self) -> str:
        return f"Crew({self.name}, status={self.status.value}, members={len(self._members)})"
